subgraphs/gmx/src/entities/protocol.ts

```diff
@@ -0,0 +1,347 @@
+export function updateProtocolOpenInterestUSD(
+  event: ethereum.Event,
+  openInterestChangeUSD: BigDecimal,
+  isIncrease: boolean
+): void {
+  const protocol = getOrCreateProtocol();
+  if (isIncrease) {
+    protocol.openInterestUSD = protocol.openInterestUSD.plus(
+      openInterestChangeUSD
+    );
+  } else {
+    protocol.openInterestUSD = protocol.openInterestUSD.minus(
+      openInterestChangeUSD
+    );
+  }
+
+  protocol._lastUpdateTimestamp = event.block.timestamp;
+  protocol.save();
+}
+
+export function incrementProtocolEventCount(
+  event: ethereum.Event,
+  eventType: EventType
+): void {
+  const protocol = getOrCreateProtocol();
+  switch (eventType) {
+    case EventType.Deposit:
+      protocol.depositCount += INT_ONE;
+      break;
+    case EventType.Withdraw:
+      protocol.withdrawCount += INT_ONE;
+      break;
+    case EventType.CollateralIn:
+      protocol.collateralInCount += INT_ONE;
+      protocol.borrowCount += INT_ONE;
+      break;
+    case EventType.CollateralOut:
+      protocol.collateralOutCount += INT_ONE;
+      break;
+    default:
+      break;
+  }
+
+  protocol.transactionCount += INT_ONE;
+  protocol._lastUpdateTimestamp = event.block.timestamp;
+  protocol.save();
+}
+
+export function incrementProtocolUniqueUsers(event: ethereum.Event): void {
+  const protocol = getOrCreateProtocol();
+  protocol.cumulativeUniqueUsers += INT_ONE;
+  protocol._lastUpdateTimestamp = event.block.timestamp;
+  protocol.save();
+}
+
+export function incrementProtocolUniqueBorrowers(event: ethereum.Event): void {
+  const protocol = getOrCreateProtocol();
+  protocol.cumulativeUniqueBorrowers += INT_ONE;
+  protocol._lastUpdateTimestamp = event.block.timestamp;
+  protocol.save();
+}
+
+export function incrementProtocolUniqueLiquidators(
+  event: ethereum.Event
+): void {
+  const protocol = getOrCreateProtocol();
+  protocol.cumulativeUniqueLiquidators += INT_ONE;
+  protocol._lastUpdateTimestamp = event.block.timestamp;
+  protocol.save();
+}
+
+export function incrementProtocolUniqueLiquidatees(
+  event: ethereum.Event
+): void {
+  const protocol = getOrCreateProtocol();
+  protocol.cumulativeUniqueLiquidatees += INT_ONE;
+  protocol._lastUpdateTimestamp = event.block.timestamp;
+  protocol.save();
+}
+
+export function incrementProtocolOpenPositionCount(
+  event: ethereum.Event,
+  positionSide: string
+): void {
+  const protocol = getOrCreateProtocol();
+  if (PositionSide.LONG == positionSide) {
+    protocol.longPositionCount += INT_ONE;
+  } else {
+    protocol.shortPositionCount += INT_ONE;
+  }
+  protocol.openPositionCount += INT_ONE;
+  protocol.cumulativePositionCount += INT_ONE;
+  protocol._lastUpdateTimestamp = event.block.timestamp;
+  protocol.save();
+}
+
+export function decrementProtocolOpenPositionCount(
+  event: ethereum.Event,
+  positionSide: string
+): void {
+  const protocol = getOrCreateProtocol();
+  if (PositionSide.LONG == positionSide) {
+    protocol.longPositionCount =
```

Comment:
Users can open a long position using the vault. e.g. to open a long, a user can deposit 1 ETH into the vault and open a position of $25,000, if the price of ETH at the time of opening the position is $5000, then this would be a 5x long position. If the price of ETH increases by 10%, the user would make a profit of $25,000 * 10% = $2500. A snapshot of the collateral is taken when the position is opened, so in this example, the collateral would be recorded as $5000 and will not change even if the price of ETH changes. 

Users can also open a short position using the vault. e.g. to open a short, a user can deposit 5000 USDC into the vault and open a position of $25,000. Stablecoins are required as collateral for shorts.